pkg/graveler/ref/manager_test.go

```diff
@@ -1226,3 +1226,122 @@ func TestManager_SetRepositoryMetadata(t *testing.T) {
+	})
+	testutil.Must(t, err)
+	ctx := context.Background()
+
+	t.Run("get_pull_request_exists", func(t *testing.T) {
+		expected := graveler.PullRequestRecord{
+			ID: "",
+			PullRequest: graveler.PullRequest{
+				CreatedAt:   time.Now().UTC(),
+				Status:      graveler.PullRequestStatus_CLOSED,
+				Title:       "some title",
+				Author:      "some author",
+				Description: "some description",
+				Source:      "dev",
+				Destination: "main",
+				CommitID:    "",
+			},
+		}
+		require.NoError(t, r.CreatePullRequest(ctx, repository, expected.ID, &expected.PullRequest))
+		pull, err := r.GetPullRequest(ctx, repository, expected.ID)
+		require.NoError(t, err)
+		require.Equal(t, expected.PullRequest, *pull)
+	})
+
+	t.Run("get_pull_request_doesnt_exists", func(t *testing.T) {
+		_, err := r.GetPullRequest(ctx, repository, "pull2")
+		require.ErrorIs(t, err, graveler.ErrPullRequestNotFound)
+	})
+}
+
+func TestManager_DeletePullRequest(t *testing.T) {
+	r, _ := testRefManager(t)
+	repository, err := r.CreateRepository(context.Background(), "repo1", graveler.Repository{
+		StorageNamespace: "s3://",
+		CreationDate:     time.Now(),
+		DefaultBranchID:  "main",
+	})
+	testutil.Must(t, err)
+	ctx := context.Background()
+
+	t.Run("delete_pull_request_exists", func(t *testing.T) {
+		rec := graveler.PullRequestRecord{
+			ID: "",
+			PullRequest: graveler.PullRequest{
+				CreatedAt:   time.Now().UTC(),
+				Status:      graveler.PullRequestStatus_CLOSED,
+				Title:       "some title",
+				Author:      "some author",
+				Description: "some description",
+				Source:      "dev",
+				Destination: "main",
+				CommitID:    "",
+			},
+		}
+		require.NoError(t, r.CreatePullRequest(ctx, repository, rec.ID, &rec.PullRequest))
+		err := r.DeletePullRequest(ctx, repository, rec.ID)
+		require.NoError(t, err)
+	})
+
+	t.Run("delete_pull_request_doesnt_exists", func(t *testing.T) {
+		// No error expected if key not found
+		require.NoError(t, r.DeletePullRequest(ctx, repository, "pull2"))
+	})
+}
+
+func TestManager_UpdatePullRequest(t *testing.T) {
+	r, _ := testRefManager(t)
+	repository, err := r.CreateRepository(context.Background(), "repo1", graveler.Repository{
+		StorageNamespace: "s3://",
+		CreationDate:     time.Now(),
+		DefaultBranchID:  "main",
+	})
+	testutil.Must(t, err)
+	ctx := context.Background()
+	expected := graveler.PullRequestRecord{
+		ID: "",
+		PullRequest: graveler.PullRequest{
+			CreatedAt:   time.Now().UTC(),
+			Status:      graveler.PullRequestStatus_CLOSED,
+			Title:       "some title",
+			Author:      "some author",
+			Description: "some description",
+			Source:      "dev",
+			Destination: "main",
+			CommitID:    "",
+		},
+	}
+
+	t.Run("update_pull_request_exists", func(t *testing.T) {
+		require.NoError(t, r.CreatePullRequest(ctx, repository, expected.ID, &expected.PullRequest))
+
+		err := r.UpdatePullRequest(ctx, repository, expected.ID, func(request *graveler.PullRequest) (*graveler.PullRequest, error) {
```

Comment:
Generally, prefer creating a new object instead of modifying an existing one.
In this case, the one that will add a new test after this will expect to have "some title" as the title of `expected`, but it'll be different.